code/clustering.py

```
@@ -0,0 +1,31 @@
+import pandas as pd
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+import matplotlib.pyplot as plt
+
+# Load embedded data
+df = pd.read_pickle("job_embeddings.pkl")
```

Ruff (0.8.2)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)